test/runtests.jl

```diff
@@ -60,6 +60,22 @@ end
             @test_throws DimensionMismatch FixedSizeArray{Int}(oa)
             @test_throws DimensionMismatch FixedSizeVector{Int}(oa)
         end
+        @testset "taken from Julia's test/core.jl" begin
+            # inspired by:
+            # https://github.com/JuliaLang/julia/blob/83929ad883c97fa4376618c4559b74f6ada7a3ce/test/core.jl#L7211-L7228
+            b = prevpow(2, typemax(Int))
+            test_inferred(FixedSizeArray{Int}, FixedSizeArray{Int,3}, (undef, 0, b, b))
+            test_inferred(FixedSizeArray{Int}, FixedSizeArray{Int,3}, (undef, b, b, 0))
+            test_inferred(FixedSizeArray{Int}, FixedSizeArray{Int,5}, (undef, b, b, 0, b, b))
+            @test_throws ArgumentError FixedSizeArray{Int}(undef, b, b)
+            @test_throws ArgumentError FixedSizeArray{Int}(undef, 1, b, b)
+            @test_throws ArgumentError FixedSizeArray{Int, 2}(undef, 0, -10)
+            @test_throws ArgumentError FixedSizeArray{Int, 2}(undef, -10, 0)
+            @test_throws ArgumentError FixedSizeArray{Int, 2}(undef, -1, -1)
```

Comment:
Do these need to use `FixedSizeArray{Int, 2}` rather than `FixedSizeArray{Int}`, like the rest of the tests?

Reply:
The parameter `2` is only here because I copied the tests from Julia's test suite. I'll remove it.